subsys/esb/esb.c

```diff
@@ -258,18 +279,21 @@ static inline void apply_errata143_workaround(void)
 	/* Load the two addresses before comparing them to ensure
 	 * defined ordering of volatile accesses.
 	 */
-	uint32_t addr0 = NRF_RADIO->BASE0 & base_address_mask;
-	uint32_t addr1 = NRF_RADIO->BASE1 & base_address_mask;
+	uint32_t addr0 = nrf_radio_base0_get(NRF_RADIO) & base_address_mask;
+	uint32_t addr1 = nrf_radio_base1_get(NRF_RADIO) & base_address_mask;
 
 	if (addr0 == addr1) {
-		uint32_t prefix0 = NRF_RADIO->PREFIX0 & 0x000000FF;
-		uint32_t prefix1 = (NRF_RADIO->PREFIX0 & 0x0000FF00) >> 8;
-		uint32_t prefix2 = (NRF_RADIO->PREFIX0 & 0x00FF0000) >> 16;
-		uint32_t prefix3 = (NRF_RADIO->PREFIX0 & 0xFF000000) >> 24;
-		uint32_t prefix4 = NRF_RADIO->PREFIX1 & 0x000000FF;
-		uint32_t prefix5 = (NRF_RADIO->PREFIX1 & 0x0000FF00) >> 8;
-		uint32_t prefix6 = (NRF_RADIO->PREFIX1 & 0x00FF0000) >> 16;
-		uint32_t prefix7 = (NRF_RADIO->PREFIX1 & 0xFF000000) >> 24;
+		uint32_t radio_prefix0 = nrf_radio_prefix0_get(NRF_RADIO);
```

Comment:
It seems that the variable type can be reduced to uint8_t.

Reply:
Yes they can be reduced also magic numbers here can be replaced by MDK definitions. Done